### PR TITLE
libcsa: init at 1.26-unstable-2024-03-22

### PR DESCRIPTION
**Pull request by mk3z**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/libcsa/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "7b48134613d1d3b337af6d5762df9999a703fb1a";
+    hash = "sha256-G/VhXpdvXBT9I6pwiQXVqCoXhc29wJQpGyLeM3kgv7I=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/csa";
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = with lib; {
+    description = "C code for cubic spline approximation of 2D scattered data.";
```
Comment (Sigmanificient) [inline pkgs/by-name/li/libcsa/package.nix:23]: Sorry to ask you again, this time with trailling period :sweat_smile: 
Check docs on  https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Review by Sigmanificient — APPROVED** [review]
Looks fine to me, very elegant derivation

**Comment by mk3z** [comment]
I have another very similar library waiting to be merged here: https://github.com/NixOS/nixpkgs/pull/325507. I was wondering whether I should name it libnn instead of nn. What is the official policy on the lib prefix? I chose to name this libcsa just because csa was already taken.

**Comment by Sigmanificient** [comment]
sadly I cannot give you an answer, but i can recommend asking on the [discourse](discourse.nixos.org)

**Review by pbsds — COMMENTED** [review]
please set meta.mainProgram

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 325787` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libcsa</li>
  </ul>
</details>

**Pull request merged**